File: Backend/Models/Storageprovider.cs
using System;
using System.Collections.Generic;

namespace SkyVault.Models;

public partial class Storageprovider
{
    public Guid Providerid { get; set; }

    public string Name { get; set; } = null!;

    public string Providertype { get; set; } = null!;

    public bool Isactive { get; set; }

    public DateTime Createdat { get; set; }

    public virtual ICollection<Storageaccount> Storageaccounts { get; set; } = new List<Storageaccount>();
}

File: Frontend/src/models/storageProvider/StorageProvider.ts
export interface CreateStorageProviderRequest {
  name: string
  providerType: string
}

export interface UpdateStorageProviderRequest {
  name: string
}

export interface StorageProviderResponse {
  providerId: string
  name: string
  providerType: string
  isActive: boolean
  createdAt: string
}
